### PR TITLE
[WIN32SS][NTGDI] Improve Text Transformation

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
- Reduce `gdi32_apitest!TextTransform` failures.
- Optimize for speed.
- Add `Width` and `Escapement` parameters to `FONT_CACHE_ENTRY` structure.

JIRA issue: [CORE-15554](https://jira.reactos.org/browse/CORE-15554)

BEFORE:
before
AFTER:
bingo
Bingo!

### CODE REVIEW

**Comment by katahiromz** [comment]
Spotify's issue is not fixed yet.
spotify-not-fixed-yet

**Comment by katahiromz** [comment]
pangotest-failure
PangoTest fails.

**Comment by HBelusca** [comment]
> PangoTest fails.

It's as if the text rotated with angles > 180 degrees and < 360 degrees only didn't display, **except for** the letter "e" (why??????)

File: win32ss/gdi/ntgdi/freetype.c [hunk 1]
```diff
@@ -5922,12 +5927,14 @@ IntExtTextOutW(
     }
     else /* TA_TOP */
     {
-        vecs[1].y = 0;                          // upper left
+        vecs[1].y = 0;                           // upper left
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/freetype.c]: extra space?

Comment (katahiromz) [reply win32ss/gdi/ntgdi/freetype.c]: Fixed.

File: win32ss/gdi/ntgdi/freetype.c [hunk 2]
```diff
@@ -5871,21 +5854,43 @@ IntExtTextOutW(
...
+        pmxWorldToDevice = DC_pmxWorldToDevice(dc);
     }
     else
     {
-        XStart64 = YStart64 = 0;
+        pmxWorldToDevice = (PMATRIX)&gmxWorldToDeviceDefault;
     }
+    FtMatrixFromMx(&matWorld, pmxWorldToDevice);
+    matWorld.xy = -matWorld.xy;
+    matWorld.yx = -matWorld.yx;
```
Comment (katahiromz) [inline win32ss/gdi/ntgdi/freetype.c]: Reverse rotation???

**Comment by katahiromz** [comment]
cba1a2f7423c0d6dbc8b3018a204d9bd266a4660
https://build.reactos.org/builders/Build%20GCCLin_x86/builds/24604

c17db672efa1471c57c4a1658414dd2788c5270e
https://build.reactos.org/builders/Build%20GCCLin_x86/builds/24605

**Comment by katahiromz** [comment]
I can see +26 improvements in gdi32:TextTransform.

According to CORE-13107, kernel32:sync is false alerm: https://jira.reactos.org/browse/CORE-13107

**Pull request merged**